### PR TITLE
Elaticsearch - 5 - improved search and filters

### PR DESCRIPTION
**Pull request by robert-bo-davis**
## Description

This is a PR in the elasticsearch chain. It should be testable.

This moves to using more complex mapping/indexing to achieve better full text search results and adds some working filter examples and tests.

### CODE REVIEW

File: packages/elasticsearch-client/src/config/default/resources/index-document.ts [hunk 1]
```diff
@@ -64,7 +74,8 @@ export const convertDocument = (resource: ReferrableResource): ResourcesIndexDoc
 
   return {
     name,
-    text1,
-    text2,
+    high_boost_global: highBoostGlobal.join(' '),
```
Comment (robert-bo-davis) [inline packages/elasticsearch-client/src/config/default/resources/index-document.ts:77]: I moved from arrays to a more common giant text string search which returns more reliable scoring/ordering, in my opinion. You'll see below that the results in tests in resources-service are ordered by documents that contain repeat values more often, which I think is likely the more desirable behavior.

File: packages/elasticsearch-client/src/config/default/resources/index-document.ts [hunk 2]
```diff
@@ -16,40 +16,50 @@
...
+  isHighBoostGlobalField,
+  isMappingField,
+  mappingFields,
+  ResourcesIndexDocument,
+} from './config';
 
+// TODO: when we have more than one index and config type, we should probably make this a little more generic
+// and just import the config to generate it. Leaving here for now.
 export const convertDocument = (resource: ReferrableResource): ResourcesIndexDocument => {
   const { name } = resource;
```
Comment (robert-bo-davis) [inline packages/elasticsearch-client/src/config/default/resources/index-document.ts:29]: In our data structure, name is a single field, but the example payload from the client has multiple languages. Will name exist both as a property of the top level resource and as a regular attribute property that may have multiple languages?

If so, this should probably be refactored to use the normal mapFields setup.

Comment (stephenhand) [reply packages/elasticsearch-client/src/config/default/resources/index-document.ts:29]: I imagine there will be both in a lot of cases. The name on the DB record is intended to be a universal, human readable handle that we can assume exists across any resource DB implementation, but other, helpline specific 'name' type attributes might well exist in the helpline's specific schema - there could be several of these, depending on the helpline's requirements (like KHP have a 'name' and and 'official name'), and they could be multilingual

**Review by stephenhand — APPROVED** [review]

File: packages/elasticsearch-client/src/config/default/resources/index-document.ts (hunk 2 above)
Comment (robert-bo-davis) [reply packages/elasticsearch-client/src/config/default/resources/index-document.ts:29]: Excellent. I'll leave as is for now and we can address the final tweaks to the mapping in one of the future ticket.

**Pull request merged**